kpops/components/base_components/base_defaults_component.py

```diff
@@ -122,25 +121,29 @@ def extend_with_defaults(cls, config: KpopsConfig, **kwargs: Any) -> dict[str, A
             elif is_dataclass(v):
                 kwargs[k] = asdict(v)
 
-        log.debug(
-            typer.style(
-                "Enriching component of type ", fg=typer.colors.GREEN, bold=False
-            )
-            + typer.style(cls.type, fg=typer.colors.GREEN, bold=True, underline=True)
-        )
-        main_default_file_path, environment_default_file_path = get_defaults_file_paths(
-            config, ENV.get("environment")
+        # TODO: Maybe there is a better way to pas pipeline_path here
+        env_pipeline_path = ENV.get("pipeline_path")
+        # TODO: One way would be add a field to KPOps Config and skip it from the JSON
+        # env_pipeline_path = config.pipeline_path
+        if not env_pipeline_path:
+            env_pipeline_path = Path()
+        pipeline_path = Path(env_pipeline_path)
+        defaults_file_paths_ = get_defaults_file_paths(
+            pipeline_path, config, ENV.get("environment")
         )
-        defaults = cls.load_defaults(
-            main_default_file_path, environment_default_file_path
+        defaults = cls.load_defaults(*defaults_file_paths_)
+        log.debug(
+            typer.style("Enriching component of type ", bold=False)
+            + typer.style(cls.type, bold=True, underline=True)
         )
         return update_nested_pair(kwargs, defaults)
 
     @classmethod
     def load_defaults(cls, *defaults_file_paths: Path) -> dict[str, Any]:
         """Resolve component-specific defaults including environment defaults.
 
-        :param *defaults_file_paths: Path to `defaults.yaml`, ordered from lowest to highest priority, i.e. `defaults.yaml`, `defaults_{environment}`.yaml
+        :param *defaults_file_paths: Path to `defaults.yaml`, ordered from highest to lowest priority,
```

Comment:
breaking change?

also in the other PR you removed the star operator (`*`) from the docs. I am not an expert at pydocs, not sure what's correct, just wanted to point it out

Reply:
Yes breaking change! 🙂 
We are totally removing the `--defaults` flag from the CLI, and the `default-path` from the config!
It was the pyCharm doc string checker giving me the warning fix and I took it!